crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -115,3 +115,38 @@ def _(flag: bool):
     # error: [unresolved-attribute] "Type `Literal[C1, C2]` has no attribute `x`"
     reveal_type(C.x)  # revealed: Unknown
 ```
+
+## Objects of all types have a `__class__` method
+
+```py
+import typing
+
+reveal_type(typing.__class__)  # revealed: Literal[ModuleType]
+
+a = 42
+reveal_type(a.__class__)  # revealed: Literal[int]
+
+b = "42"
+reveal_type(b.__class__)  # revealed: Literal[str]
+
+c = b"42"
+reveal_type(c.__class__)  # revealed: Literal[bytes]
+
+d = True
+reveal_type(d.__class__)  # revealed: Literal[bool]
+
+e = (42, 42)
+reveal_type(e.__class__)  # revealed: Literal[tuple]
+
+def f(a: int, b: typing.LiteralString, c: type[str], d: int | str):
+    reveal_type(a.__class__)  # revealed: type[int]
+    reveal_type(b.__class__)  # revealed: Literal[str]
+    reveal_type(c.__class__)  # revealed: type[type]
```

Comment:
(on `main` we infer `Literal[type]` here, which is incorrect: `c` could be a subclass of `str` with a custom metaclass. The most we can infer is that `c.__class__` is some (non-strict) subclass of `type`)

Reply:
actually I'll add this as a comment